Extension/src/LanguageServer/settings.ts

```diff
@@ -68,55 +68,105 @@ export class CppSettings extends Settings {
         super("C_Cpp", resource);
     }
 
+    private get LLVMExtension(): string {
+        return os.platform() === "win32" ? ".exe" : "";
+    }
+
+    private get clangFormatStr(): string {
+        return "clang-format";
+    }
+
+    private get clangTidyStr(): string {
+        return "clang-tidy";
+    }
+
     private get clangFormatName(): string {
-        return "clang-format" + (os.platform() === "win32" ? ".exe" : "");
+        return this.clangFormatStr + this.LLVMExtension;
+    }
+
+    private get clangTidyName(): string {
+        return this.clangTidyStr + this.LLVMExtension;
+    }
+
+    public get clangTidyPath(): string | undefined {
+        return this.getClangPath(false);
     }
 
     public get clangFormatPath(): string | undefined {
-        let path: string | undefined | null = super.Section.get<string>("clang_format_path");
+        return this.getClangPath(true);
+    }
+
+    private getClangPath(isFormat: boolean): string | undefined {
+        let path: string | undefined | null = super.Section.get<string>(isFormat ? "clang_format_path" : "codeAnalysis.clangTidy.path");
         if (!path) {
-            const cachedClangFormatPath: string | null | undefined = getCachedClangFormatPath();
-            if (cachedClangFormatPath !== undefined) {
-                if (cachedClangFormatPath === null) {
+            const cachedClangPath: string | null | undefined = isFormat ? getCachedClangFormatPath() : getCachedClangTidyPath();
+            if (cachedClangPath !== undefined) {
+                if (cachedClangPath === null) {
                     return undefined;
                 }
-                return cachedClangFormatPath;
+                return cachedClangPath;
             }
-            path = which.sync('clang-format', { nothrow: true });
-            setCachedClangFormatPath(path);
+            const clangStr: string = isFormat ? this.clangFormatStr : this.clangTidyStr;
+            const clangName: string = isFormat ? this.clangFormatName : this.clangTidyName;
+            const setCachedClangPath: (path: string | null) => void = isFormat ? setCachedClangFormatPath : setCachedClangTidyPath;
+            path = which.sync(clangName, { nothrow: true });
+            setCachedClangPath(path);
             if (!path) {
                 return undefined;
             } else {
-                // Attempt to invoke both our own version of clang-format to see if we can successfully execute it, and to get it's version.
-                let clangFormatVersion: string;
+                // Attempt to invoke both our own version of clang-* to see if we can successfully execute it, and to get it's version.
```

Comment:
Typo: it's -> its

Reply:
Thanks. Fixed with https://github.com/microsoft/vscode-cpptools/pull/7816/commits/374060eacf860d77a89a3e32f94b65f6175dc82b